gen/src/Gen/Output/Template/CabalFile.hs

```diff
@@ -1,16 +1,28 @@
-module Gen.Output.Template.CabalFile where
+module Gen.Output.Template.CabalFile
+  ( CabalFile (..),
+    arguments,
+    fromLibrary,
+  )
+where
 
 import Data.Aeson (toJSON)
 import qualified Data.Aeson as Aeson
 import qualified Data.HashMap.Strict as HashMap
 import Gen.Prelude
+import Gen.Types.Config (Library)
+import qualified Gen.Types.Config as Config
+import Gen.Types.NS (unNS)
+import Gen.Types.Service (metadata, service)
+import qualified Gen.Types.Service as Service
 
+-- | Arguments for the @cabal.ede@ template.
 data CabalFile = CabalFile
```

Comment:
Do you think the version of cabal that is described in the generated file should be a property of this datatype or of the template itself?

Reply:
Nah it's fine. It's just a file format specifier and if we start depending on new features we'll be changing the template structure anyway. No point putting `"2.2"` in our code forever.